lib/request_overrider.js

```diff
@@ -52,7 +56,8 @@ function setRequestHeaders(req, options, interceptor) {
   //  We mock request headers if these were specified.
   if (interceptor.reqheaders) {
     _.forOwn(interceptor.reqheaders, function(val, key) {
-      setHeader(req, key, val);
+      if (!_.isFunction(val))
```

Comment:
`val` can be the matching function from `reqheaders`.  During the matching phase, headers specified in `reqheaders` are copied to the client request object - now using the setHeader() function. setHeader() checks for valid characters in the header value, so it errors when the value is a function.

I am actually not sure why the headers are copied from the interceptor to the request object, since I thought reqheaders are used only for matching. In any case, setting the function as a header value seems unintended, and taking it out did not break any tests.